host/tests/dsClasses/dsDisplay.py

```diff
@@ -0,0 +1,251 @@
+# * See the License for the specific language governing permissions and
+# * limitations under the License.
+# *
+#* ******************************************************************************
+
+import yaml
+import os
+import sys
+from enum import Enum, auto
+
+# Add parent outside of the class directory
+dir_path = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(dir_path+"/../")
+
+from raft.framework.plugins.ut_raft.configRead import ConfigRead
+from raft.framework.plugins.ut_raft.utSuiteNavigator import UTSuiteNavigatorClass
+from raft.framework.plugins.ut_raft.interactiveShell import InteractiveShell
+
+class dsVideoPortType(Enum):
+    dsVIDEOPORT_TYPE_RF     = 0
+    dsVIDEOPORT_TYPE_BB =auto()
+    dsVIDEOPORT_TYPE_SVIDEO =auto()
+    dsVIDEOPORT_TYPE_1394 =auto()
+    dsVIDEOPORT_TYPE_DVI =auto()
+    dsVIDEOPORT_TYPE_COMPONENT =auto()
+    dsVIDEOPORT_TYPE_HDMI =auto()
+    dsVIDEOPORT_TYPE_HDMI_INPUT =auto()
+    dsVIDEOPORT_TYPE_INTERNAL =auto()
+
+class dsDisplayClass():
+
+    moduleName = "dsDisplay"
+    menuConfig =  dir_path + "/dsDisplay_test_suite.yml"
+    testSuite = "L3 dsDisplay"
+
+    """
+    Device Settings Display Class
+    This module provides common extensions for device Settings Display Module.
+    """
+    def __init__(self, deviceProfilePath:str, session=None ):
+        """
+        Initializes the dsDisplay class function.
+        """
+        self.deviceProfile = ConfigRead( deviceProfilePath, self.moduleName)
+        self.utMenu        = UTSuiteNavigatorClass(self.menuConfig, self.moduleName, session)
+
+        self.utMenu.start()
+
+    def initialise(self, device_type:int=0):
+        """
+        Initializes the device settings Display module.
+
+        Args:
+            device_type (int, optional): 0 - sink device, 1 - source device. Defaults to sink.
+
+        Returns:
+                None
+        """
+        result = self.utMenu.select( self.testSuite, "Initialize dsDisplay")
+
+    def getEdid(self, video_port: str, port_index: int = 0):
+        """
+        Gets the EDID information.
+
+        Args:
+            port_type (str): name of the video port. Refer dsVideoPortType enum
+            port_index (int, optional): port index. Defaults to 0
+
+            Returns:
+                None
+        """
+        promptWithAnswers = [
+                {
+                    "query_type": "list",
+                    "query": "Select dsVideo Port:",
+                    "input":str(video_port)
+                },
+                {
+                    "query_type": "direct",
+                    "query": "Select dsVideo Port Index[0-10]:",
+                    "input": str(port_index)
+                }
+        ]
+
+        promptWithAnswers[0]["input"] = str(video_port)
+        promptWithAnswers[1]["input"] = str(port_index)
+
+        result = self.utMenu.select(self.testSuite, "Get display EDID", promptWithAnswers)
```

Comment:
this is get function and anyone calling this function should receive the edid data, but this function doesn't return anything. Similarly check for all get function